### ISSUE TITLE
Wrong error when compiling code containing IEnumerable<Span<T>>

### ISSUE THREAD

**Issue by aannenko** [dotnet/roslyn#76007]
**Version Used**:
9.0.100

**Steps to Reproduce**:
1. `dotnet new console`
2. Replace contents of the `Program.cs` with this:
```csharp
foreach (var item in IntsAsStrings(5))
    Console.WriteLine(item.ToString());

// intentionally invalid code
static IEnumerable<ReadOnlySpan<char>> IntsAsStrings(int count)
{
    var span = new char[11];
    for (var i = 0; i < count; i++)
    {
        i.TryFormat(span, out var charsWritten);
        yield return span.AsSpan(0, charsWritten);
    }
}
```
3. Try compiling it.

**Diagnostic Id**:
| Severity | Code | Description | Project | File | Line | Suppression State |
|---|---|---|---|---|---|---|
| Error (active) | MSB6006 | "csc.exe" exited with code -2146232797. | test | C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets | 89 | |

**Expected Behavior**:
Like in .NET 8, we get an appropriate error:

CS9244 The type 'ReadOnlySpan<char>' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'T' in the generic type or method 'IEnumerable<T>'

**Actual Behavior**:
csc.exe tries and fails to compile the above code.

**Comment by jjonescz** [comment]
Looks like a duplicate of https://github.com/dotnet/roslyn/issues/75569.

**Comment by aannenko** [comment]
Yes, thanks. Should I close this one then?
